app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ComposeAbstractTabViewHolder.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray.browser.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.recyclerview.widget.RecyclerView
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.tabstray.TabViewHolder
+import org.mozilla.fenix.compose.ComposeViewHolder
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * [RecyclerView.ViewHolder] used for Jetpack Compose UI content .
+ *
+ * @param composeView [ComposeView] which will be populated with Jetpack Compose UI content.
+ * @param viewLifecycleOwner [LifecycleOwner] life cycle owner for the view.
+ */
+abstract class ComposeAbstractTabViewHolder(
+    val composeView: ComposeView,
+    val viewLifecycleOwner: LifecycleOwner
+) : TabViewHolder(composeView) {
+
+    /**
+     * Composable that contains the content for a specific [ComposeViewHolder] implementation.
+     */
+    @Composable
+    abstract fun Content(tab: TabSessionState)
+
+    fun bind(tab: TabSessionState) {
+        composeView.setViewCompositionStrategy(
+            ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed
+        )
```

Comment:
We probably don't need to set a custom strategy and can leverage the default which would have the composition destroyed when the ViewHolder is recycled.
Otherwise since we're injecting state from outside through `updateSelectedTabIndicator` and the likes this will leak to the newly wrapped composable that would get to use a recycled ViewHolder.
Can be tested by just selecting some items -> deleting them -> tap Undo -> tap one of the same items.